src/components/search/search.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import classes from './search.module.scss';
+
+interface ISearch {
+  textSearch: string;
+}
+
+export default class Search extends React.Component<object, ISearch> {
+  constructor(props: object) {
+    super(props);
+    this.state = {
+      textSearch: localStorage.getItem('LocalStorageSearch') || '',
+    };
+    this.changeText = this.changeText.bind(this);
+  }
+
+  changeText(e: React.ChangeEvent<HTMLInputElement>): void {
+    this.setState({
+      textSearch: e.target.value,
+    });
+  }
+
+  public componentDidUpdate(): void {
+    localStorage.setItem('LocalStorageSearch', `${this.state.textSearch}`);
+  }
+
+  render() {
+    const value = this.state.textSearch;
+    return (
+      <div className={classes.search}>
+        <input id="search" type="text" value={value} onChange={this.changeText} />
```

Comment:
it is not really needed to create a variable just for 1 value, u can use like this:
```
<input
   id="search"
   type="text"
   value={this.state.textSearch}
   onChange={this.changeText}
/>
```